src/test/ui/fmt/format-args-capture-issue-93378.stderr

```diff
@@ -0,0 +1,22 @@
+error: invalid reference to positional arguments 1 and 2 (there is 1 argument)
+  --> $DIR/format-args-capture-issue-93378.rs:5:26
+   |
+LL |     println!("{a} {b} {} {} {c} {}", c = "c");
+   |                          ^^     ^^
+   |
+   = note: positional arguments are zero-based
+
+error: invalid reference to positional argument 0 (no arguments were given)
+  --> $DIR/format-args-capture-issue-93378.rs:9:23
+   |
+LL |     println!("{a:.n$} {b:.*}");
+   |               ------- ^^^--^
+   |               |          |
+   |               |          this precision flag adds an extra required argument at position 0, which is why there are 3 arguments expected
```

Comment:
This output needs some work to make it clearer :(

Reply:
Yes, there's quite a few cases right now where this macro produces suboptimal diagnostics.

We have plans to change how `fmt::Arguments` is implemented/represented, which would require a lot of changes to the `format_args` builtin macro. That would probably be the best time to just rewrite the macro and the way it produces diagnostics.